components/console/esp_linenoise.c

```diff
@@ -0,0 +1,1894 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
```

Comment:
To reflect the copyrights and the actual license below, suggest:

```suggestion
 * SPDX-FileCopyrightText: 2010-2016 Salvatore Sanfilippo <antirez at gmail dot com>
 * SPDX-FileCopyrightText: 2010-2013 Pieter Noordhuis <pcnoordhuis at gmail dot com>
 * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
 *
 * SPDX-License-Identifier: BSD-2-Clause
```